src/wp-includes/blocks/comment-template.php

```diff
@@ -35,16 +35,19 @@ function block_core_comment_template_render_comments( $comments, $block ) {
 		 * We set commentId context through the `render_block_context` filter so
 		 * that dynamically inserted blocks (at `render_block` filter stage)
 		 * will also receive that context.
+		 *
+		 * Use an early priority to so that other 'render_block_context' filters
+		 * have access to the values.
 		 */
-		add_filter( 'render_block_context', $filter_block_context );
+		add_filter( 'render_block_context', $filter_block_context, 0 );
```

Comment:
One minor note: The [docs](https://developer.wordpress.org/plugins/hooks/actions/#priority) say that 

> priorities are positive integers, typically between 1 and 20

```suggestion
		add_filter( 'render_block_context', $filter_block_context, 1 );
```